cmake/CMakeLists.txt

```diff
@@ -0,0 +1,160 @@
+check_include_files(unistd.h         HAVE_UNISTD_H)
+check_include_files("stdlib.h;stdarg.h;string.h;float.h" STDC_HEADERS)
+
+check_symbol_exists(EFD_CLOEXEC     "sys/eventfd.h" HAVE_DECL_EFD_CLOEXEC)
+check_symbol_exists(EFD_NONBLOCK    "sys/eventfd.h" HAVE_DECL_EFD_NONBLOCK)
+check_symbol_exists(TFD_CLOEXEC     "sys/eventfd.h" HAVE_DECL_TFD_CLOEXEC)
+check_symbol_exists(timerfd_create  "sys/timerfd.h" HAVE_TIMERFD) 
+
+check_struct_has_member("struct timespec" tv_sec time.h HAVE_STRUCT_TIMESPEC)
+
+if(BUILD_SHARED_LIBS)
+  set(libusb_BUILD_SHARED_LIBS_DEFAULT ON)
+else (BUILD_SHARED_LIBS)
+  set(libusb_BUILD_SHARED_LIBS_DEFAULT OFF)
+endif (BUILD_SHARED_LIBS)
+
+option(libusb_BUILD_SHARED_LIBS "Build Shared Libraries for libusb" ${libusb_BUILD_SHARED_LIBS_DEFAULT})
+option(libusb_BUILD_TESTING "Build Tests" ON)
+option(ENABLE_LOGGING "Enable Logging" ON)
+
+if(WIN32)
+    set(PLATFORM_WINDOWS 1 CACHE INTERNAL "")
+endif()
+if(UNIX)
+    set(PLATFORM_POSIX 1  CACHE INTERNAL "")
+endif()
+
+if(("${CMAKE_CXX_COMPILER_ID}" STREQUAL Clang) OR ("${CMAKE_CXX_COMPILER_ID}" STREQUAL GNU))
+    check_c_compiler_flag("-fvisibility=hidden" HAVE_VISIBILITY)
+endif()
+
+if(HAVE_VISIBILITY)
+    set(DEFAULT_VISIBILITY "__attribute__((visibility(\"default\")))"  CACHE INTERNAL "")
+else()
+    set(DEFAULT_VISIBILITY ""  CACHE INTERNAL "")
+endif()
+
+configure_file(config.h.in config.h @ONLY)
+
+set(libusb_ROOT ../libusb)
+
+set(common_sources
+    "${libusb_ROOT}/core.c"
+    "${libusb_ROOT}/descriptor.c"
+    "${libusb_ROOT}/hotplug.c"
+    "${libusb_ROOT}/io.c"
+    "${libusb_ROOT}/sync.c"
+    "${libusb_ROOT}/strerror.c"
+
+    "${libusb_ROOT}/libusb.h"
+    "${libusb_ROOT}/libusbi.h"
+    "${libusb_ROOT}/version_nano.h"
+    "${libusb_ROOT}/version.h"
+
+    "${CMAKE_CURRENT_BINARY_DIR}/config.h"
+)
+
+if(libusb_BUILD_SHARED_LIBS)
+    add_library(usb-1.0 SHARED ${common_sources})
+else()
+    add_library(usb-1.0 STATIC ${common_sources})
+endif()
+
+target_include_directories(usb-1.0 PUBLIC "${libusb_ROOT}")
+target_include_directories(usb-1.0 PUBLIC "${CMAKE_CURRENT_BINARY_DIR}")
+target_include_directories(usb-1.0 PRIVATE "${libusb_ROOT}/os")
+
+if(WIN32)
+    target_sources(usb-1.0 PRIVATE
+        "${libusb_ROOT}/os/windows_common.c"
+        "${libusb_ROOT}/os/windows_usbdk.c"
+        "${libusb_ROOT}/os/windows_winusb.c"
+        "${libusb_ROOT}/os/events_windows.c"
+        "${libusb_ROOT}/os/threads_windows.c"
+
+        "${libusb_ROOT}/os/events_windows.h"
+        "${libusb_ROOT}/os/threads_windows.h"
+        "${libusb_ROOT}/os/windows_common.h"
+        "${libusb_ROOT}/os/windows_usbdk.h"
+        "${libusb_ROOT}/os/windows_winusb.h"
+
+        "${libusb_ROOT}/libusb-1.0.def"
+    )
+    if(MSVC)
+        target_sources(usb-1.0 PRIVATE "${libusb_ROOT}/libusb-1.0.rc")
+        target_compile_options(usb-1.0 PRIVATE /await /Zc:twoPhase- /permissive)
+    endif()
+    target_link_libraries(usb-1.0 PRIVATE windowsapp)
```

Comment:
`windowsapp` - is it used by libusb? I don't believe there are any usages of C++/WinRT

Reply:
I'll have to refresh my memory but i believe this was needed to fix some resolved references when linking the dll with mingw64.
Will add additional comments

Reply:
This [doesn't look like a mingw](https://docs.microsoft.com/en-us/uwp/win32-and-com/win32-apis) thing. Likely another artifact of the WinRT backend you've mentioned.